corehq/apps/hqwebapp/static/hqwebapp/js/components/rich_text_knockout_bindings.js

```diff
@@ -21,6 +21,44 @@ Quill.register({
     "formats/header": Header,
 });
 
+const parser = new DOMParser();
+
+const editorImages = new WeakMap();
+
+function extractImages(html) {
+    const images = new Set();
+    if (!html) {
+        return images;
+    }
+
+    const doc = parser.parseFromString(html, 'text/html');
+    const imgElements = doc.querySelectorAll('img');
+
+    imgElements.forEach(img => {
+        const src = img.getAttribute('src');
+        if (src) {
+            images.add(src);
+        }
+    });
+
+    return images;
+}
+
+function deleteUnusedImages(editor, newImages) {
+    const oldImages = editorImages.get(editor) || new Set();
```

Comment:
Are these all going to be images uploaded to HQ through the editor?  Like, could I make an email referencing an image that's hosted elsewhere?